gapic/templates/tests/unit/gapic/%name_%version/%sub/test_%service.py.j2

```diff
@@ -556,58 +614,63 @@ def test_{{ service.client_name|snake_case }}_get_mtls_endpoint_and_cert_source(
     {{ service.client_name }}
     {% endif %}
 ])
-@mock.patch.object({{ service.client_name }}, "DEFAULT_ENDPOINT", modify_default_endpoint({{ service.client_name }}))
+@mock.patch.object({{ service.client_name }}, "DEFAULT_ENDPOINT_TEMPLATE", modify_default_endpoint_template({{ service.client_name }}))
 {% if 'grpc' in opts.transport %}
-@mock.patch.object({{ service.async_client_name }}, "DEFAULT_ENDPOINT", modify_default_endpoint({{ service.async_client_name }}))
+@mock.patch.object({{ service.async_client_name }}, "DEFAULT_ENDPOINT_TEMPLATE", modify_default_endpoint_template({{ service.async_client_name }}))
 {% endif %}
 def test_{{ service.client_name|snake_case }}_client_api_endpoint(client_class):
     {# TODO(clean-up): remove redundant tests that are already covered by the smaller functions tests #}
     mock_client_cert_source = client_cert_source_callback
+    api_override = "foo.com"
+    default_universe = {{ service.client_name }}._DEFAULT_UNIVERSE
+    default_endpoint = {{ service.client_name }}.DEFAULT_ENDPOINT_TEMPLATE.format(UNIVERSE_DOMAIN=default_universe)
+    mock_universe = "bar.com"
+    mock_endpoint = {{ service.client_name }}.DEFAULT_ENDPOINT_TEMPLATE.format(UNIVERSE_DOMAIN=mock_universe)
 
     # If ClientOptions.api_endpoint is set and GOOGLE_API_USE_CLIENT_CERTIFICATE="true",
     # use ClientOptions.api_endpoint as the api endpoint regardless.
     with mock.patch.dict(os.environ, {"GOOGLE_API_USE_CLIENT_CERTIFICATE": "true"}):
-        mock_api_endpoint = "foo"
-        options = client_options.ClientOptions(client_cert_source=mock_client_cert_source, api_endpoint=mock_api_endpoint)
-        client = client_class(client_options=options, credentials=ga_credentials.AnonymousCredentials())
-        assert client.api_endpoint == mock_api_endpoint
-
-    # If ClientOptions.api_endpoint is set and GOOGLE_API_USE_CLIENT_CERTIFICATE="false",
-    # use ClientOptions.api_endpoint as the api endpoint.
-    with mock.patch.dict(os.environ, {"GOOGLE_API_USE_CLIENT_CERTIFICATE": "false"}):
-        mock_api_endpoint = "foo"
-        options = client_options.ClientOptions(client_cert_source=mock_client_cert_source, api_endpoint=mock_api_endpoint)
-        client = client_class(client_options=options, credentials=ga_credentials.AnonymousCredentials())
-        assert client.api_endpoint == mock_api_endpoint
+        options = client_options.ClientOptions(client_cert_source=mock_client_cert_source, api_endpoint=api_override)
```

Comment:
It seems like we're testing combinations of 3 paramters, `GOOGLE_API_USE_CLIENT_CERTIFICATE` `(true` or `false`), `GOOGLE_API_USE_MTLS_ENDPOINT` (`never`, `always`, `auto`) and `ClientOptions.api_endpoint` ( set or unset) for 2×3×2=12 combinations. We should list and test them systematically. For those combinations that have the same behavior, we could just choose one representative test case and add a comment about the other cases expected to behave the same way